include/boost/geometry/algorithms/detail/convert_units.hpp

```diff
@@ -0,0 +1,177 @@
+struct convert_coordinate
+{};
+
+template <>
+struct convert_coordinate<degree, radian>
+{
+    template <typename CoordinateType>
+    static inline CoordinateType apply(CoordinateType const& value)
+    {
+        return value * math::d2r<CoordinateType>();
+    }
+};
+
+template <>
+struct convert_coordinate<radian, degree>
+{
+    template <typename CoordinateType>
+    static inline CoordinateType apply(CoordinateType const& value)
+    {
+        return value * math::r2d<CoordinateType>();
+    }
+};
+
+
+template
+<
+    typename UnitsFrom,
+    typename UnitsTo,
+    bool SameUnits = boost::is_same<UnitsFrom, UnitsTo>::value
+>
+struct convert_point
+{
+    template <std::size_t Dimension, typename Point>
+    static inline void convert_units(Point& point)
+    {
+        geometry::set
+            <
+                Dimension
+            >(point,
+              convert_coordinate
+                  <
+                      UnitsFrom, UnitsTo
+                  >::apply(geometry::get<Dimension>(point)));
+    }
+
+    template <typename Point>
+    static inline void apply(Point& point)
+    {
+        convert_units<0>(point);
+        convert_units<1>(point);
+    }
+};
+
+template <typename UnitsFrom, typename UnitsTo>
+struct convert_point<UnitsFrom, UnitsTo, true>
+{
+    template <typename Point>
+    static inline void apply(Point&)
+    {
+    }
+};
+
+
+
+template
+<
+    typename UnitsFrom,
+    typename UnitsTo,
+    bool SameUnits = boost::is_same<UnitsFrom, UnitsTo>::value
+>
+struct convert_box
+{
+    template <typename Box>
+    static inline void apply(Box& box)
+    {
+        detail::indexed_point_view<Box, min_corner> p_min(box);
+        detail::indexed_point_view<Box, max_corner> p_max(box);
+
+        convert_point<UnitsFrom, UnitsTo>::apply(p_min);
+        convert_point<UnitsFrom, UnitsTo>::apply(p_max);
+    }
+};
+
+template <typename UnitsFrom, typename UnitsTo>
+struct convert_box<UnitsFrom, UnitsTo, true>
+{
+    template <typename Box>
+    static inline void apply(Box&)
+    {
+    }
+};
+
+
+}} // namespace detail::units_conversion
+#endif // DOXYGEN_NO_DETAIL
+
+
+#ifndef DOXYGEN_NO_DISPATCH
+namespace dispatch
+{
+
+
+template
+<
+    typename UnitsFrom,
+    typename UnitsTo,
+    typename Geometry,
+    typename Tag = typename tag<Geometry>::type
+>
+struct convert_units
+{};
+
+template <typename UnitsFrom, typename UnitsTo, typename Point>
+struct convert_units<UnitsFrom, UnitsTo, Point, point_tag>
+    : detail::units_conversion::convert_point<UnitsFrom, UnitsTo>
+{};
+
+template <typename UnitsFrom, typename UnitsTo, typename Box>
+struct convert_units<UnitsFrom, UnitsTo, Box, box_tag>
+    : detail::units_conversion::convert_box<UnitsFrom, UnitsTo>
+{};
+
+
+} // namespace dispatch
+#endif // DOXYGEN_NO_DISPATCH
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail
+{
+
+
+template <typename UnitsFrom, typename UnitsTo, typename Geometry>
+inline void convert_units(Geometry& geometry)
```

Comment:
Yes, it's also my understanding that we need a tool other than `convert()`. I'm complaining about the name, parameters and semantics in general. In Geometry functions that converts or transforms one geometry into another takes 2 parameters. This function corrects the geometry, not converts (in a Geometry sense) so a better name would be `correct_units()`. Then the 2 additional template parameters defining the units doesn't fit. At least one of them isn't needed since the Geometry is known. But for now we could leave it as it is since I don't have a better idea right now, it's a detail after all. Or do you have?